ivy/stateful/layers.py

```diff
@@ -1530,8 +1530,59 @@ def _forward(self, inputs):
             self._padding,
             data_format=self._data_format,
         )
+    
+    class MaxPool3D(Module):
+        def __init__(
+            self,
+            kernel_size,
+            stride,
+            padding,
+            /,
+            *,
+            data_format="NDHWC",
+            device=None,
+            v=None,
+            dtype=None,
+        ):
+            """
+            Class for applying Max Pooling over a mini-batch of inputs.
+            Parameters
+            ----------
+            kernel_size
+                The size of the window to take a max over.
+            stride
+                The stride of the window. Default value: 1
+            padding
+                Implicit zero padding to be added on both sides.
+            device
+                device on which to create the layer's variables 'cuda:0', 'cuda:1', 'cpu'
+            """
+            self._kernel_size = kernel_size
+            self._stride = stride
+            self._padding = padding
+            self._data_format = data_format
+            Module.__init__(self, device=device, dtype=dtype)
+
+        def _forward(self, inputs):
+            """
+            Forward pass of the layer.
+            Parameters
+            ----------
+            inputs
+                The input to the layer.
+            Returns
+            -------
+            The output of the layer.
+            """
+            return ivy.max_pool3d(
+                inputs,
+                self._kernel_size,
+                self._stride,
+                self._padding,
+                data_format=self._data_format,
+            )
```

Comment:
There is an error that ivy has no attribute MaxPool3D. You should be seeing this when you run the test locally. This is because you have defined the class inside the MaxPool2d class.